components/eam/src/physics/crm/cospsimulator_intr.F90

```diff
@@ -0,0 +1,3184 @@
+       do iz = 1,crm_nz
+         do iy = 1,crm_ny_rad
+           do ix = 1,crm_nx_rad
+             do i = 1,ncol
+               j = _IDX321(i, ix, iy, ncol, crm_nx_rad, crm_ny_rad)
+               k = pver - iz + 1
+               reff_cosp(j,k,I_LSCLIQ) = crm_rel(i,ix,iy,iz) * 1e-6_r8  ! microns to meters
+               reff_cosp(j,k,I_LSCICE) = crm_rei(i,ix,iy,iz) * 1e-6_r8  ! microns to meters
+               reff_cosp(j,k,I_LSRAIN) = 0._r8
+               reff_cosp(j,k,I_LSSNOW) = 0._r8
+               reff_cosp(j,k,I_CVCLIQ) = 0._r8
+               reff_cosp(j,k,I_CVCICE) = 0._r8
+               reff_cosp(j,k,I_CVRAIN) = 0._r8
+               reff_cosp(j,k,I_CVSNOW) = 0._r8
+               reff_cosp(j,k,I_LSGRPL) = 0._r8
+             end do
+           end do
+         end do
+       end do
+    else
+       ! Previously, I had set use_reff=.false.
+       ! use_reff = .false.  !! if you use this,all sizes use DEFAULT_LIDAR_REFF = 30.0e-6 meters
+       ! The specification of reff_cosp now follows e-mail discussion with Yuying in January 2011. (see above)
+       ! All of the values that I have assembled in the code are in microns...convert to meters since that is what COSP wants.
+       call pbuf_get_field(pbuf, rel_idx, rel  )
+       call pbuf_get_field(pbuf, rei_idx, rei)
+       ! added some more sizes to physics buffer in stratiform.F90 for COSP inputs
+       call pbuf_get_field(pbuf, lsreffrain_idx, ls_reffrain  )
+       call pbuf_get_field(pbuf, lsreffsnow_idx, ls_reffsnow  )
+       call pbuf_get_field(pbuf, cvreffliq_idx,  cv_reffliq   )
+       call pbuf_get_field(pbuf, cvreffice_idx,  cv_reffice   )
+       reff_cosp(1:ncol,1:pver,1:nhydro) = 0._r8
+       reff_cosp(1:ncol,1:pver,I_LSCLIQ) = rel(1:ncol,1:pver)*1.e-6_r8          ! (same as effc and effliq in stratiform.F90)
+       reff_cosp(1:ncol,1:pver,I_LSCICE) = rei(1:ncol,1:pver)*1.e-6_r8          ! (same as effi and effice in stratiform.F90)
+       reff_cosp(1:ncol,1:pver,I_LSRAIN) = ls_reffrain(1:ncol,1:pver)*1.e-6_r8  ! (calculated in cldwat2m_micro.F90, passed to stratiform.F90)
+       reff_cosp(1:ncol,1:pver,I_LSSNOW) = ls_reffsnow(1:ncol,1:pver)*1.e-6_r8  ! (calculated in cldwat2m_micro.F90, passed to stratiform.F90)
+       reff_cosp(1:ncol,1:pver,I_CVCLIQ) = cv_reffliq(1:ncol,1:pver)*1.e-6_r8   ! (calculated in stratiform.F90, not actually used in radiation)
+       reff_cosp(1:ncol,1:pver,I_CVCICE) = cv_reffice(1:ncol,1:pver)*1.e-6_r8   ! (calculated in stratiform.F90, not actually used in radiation)
+       reff_cosp(1:ncol,1:pver,I_CVRAIN) = ls_reffrain(1:ncol,1:pver)*1.e-6_r8  ! (same as stratiform per Andrew)
+       reff_cosp(1:ncol,1:pver,I_CVSNOW) = ls_reffsnow(1:ncol,1:pver)*1.e-6_r8  ! (same as stratiform per Andrew)
+       reff_cosp(1:ncol,1:pver,I_LSGRPL) = 0._r8                                ! (using radar default reff)
+    end if  ! use_MMF
+
+    ! Set optical depth and emissivity
+    if (use_MMF) then
```

Comment:
I noticed you left in this logic to change behavior for MMF. The more I look at this I think it kinda makes sense to have a MMF specific version, for the same reasons we have a separate radiation interface. We might need logic blocks like this anyway when we get to the final implementation of P3, so removing this non-MMF stuff might help clean it up down the line.

Reply:
Note that this IS the MMF specific version. I actually wanted to keep it generic for the time being though, so that I could easily move these changes over to the non-MMF version as well if we decide to do so. I vote we remove these logicals in a subsequent PR depending on our needs for P3, but I can strip this down here if you insist.

Reply:
sounds like a good plan.